traversal/src/main/scala/overflowdb/traversal/InitialTraversal.scala

```diff
@@ -0,0 +1,38 @@
+package overflowdb.traversal
+
+import overflowdb.Graph
+
+class InitialTraversal[+A <: overflowdb.Node] private (graph: Graph,
+                                                       label: String,
+                                                       iter: ArrayListIter[A])
+    extends Traversal[A](iter) {
+
+  def getByIndex(key: String, value: Any): Traversal[A] = {
+    if (iter.idx == 0 && graph.indexManager.isIndexed(key)) {
+      val nodes = graph.indexManager.lookup(key, value)
+      Traversal.from(nodes.iterator()).asInstanceOf[Traversal[A]]
+    } else {
+      null
```

Comment:
can we plesae avoi `null` here? I don't want to check for `null` all over the place like in old style java. `None` might be the best option here (pun intended)

Reply:
This interface is not user-facing, and None doesn't fit the desired type.

We already have a user-facing interface for this functionality (based on TraversalSource).

Reply:
> This interface is not user-facing

I am also a user


> None doesn't fit the desired type.

I don't follow. This method either returns a traversal, or not. Ever since java 8 and scala 2.x we can thankfully encapsulate this in the return type, in this case e.g. `Option[Traversal[A]]`. If we pass around `null` we have to check it all over the place, and it defers from common practices, not only for scala but also for java these days.

Reply:
This is not intended to be a widely used public API; this function has exactly one intended use-site (in the generated code) that immediately checks the result for nullity, so there is no issue about forgetting a check or "passing around `null`".

If this specific case is important to you, then I can change it?

Reply:
thank you, yes please change it.